lib/document.js

```diff
@@ -2853,19 +2853,21 @@ Document.prototype.validateSync = function(pathsToValidate, options) {
   }
   const validating = {};
 
-  paths.forEach(function(path) {
+  for (let i = 0, len = paths.length; i < len; ++i) {
+    const path = paths[i];
+
     if (validating[path]) {
-      return;
+      continue;
     }
 
     validating[path] = true;
```

Comment:
why do we need to keep track if we are validated a path already?

Reply:
I don't recall exactly, but it has something to do with subdocuments or nested paths. Having both the parent path and child path modified at the same time can cause double validation.

Reply:
God bless unit tests.
In an ideal world, we can _not_ do that and see what test fails.